### PR TITLE
Presets

### PR DESCRIPTION
**Pull request by dbolotin**

### CODE REVIEW

File: src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt [hunk 1]
```diff
@@ -37,758 +35,743 @@ import com.milaboratory.core.sequence.NucleotideSequence
...
-        set(value) {
-            warn(
-                "Don't use --chains option on the alignment step. See --chains parameter in exportAlignments and " +
-                        "exportClones actions to limit output to a subset of receptor chains."
-            )
-            field = value
-        }
+        @Option(
+            description = ["Copy original reads (sequences + qualities + descriptions) to .vdjca file."],
+            names = ["-g", "--save-reads"],
```
Comment (gnefedev) [inline src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt:180]: you can use Path type

File: src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt [hunk 2]
```diff
@@ -37,758 +35,743 @@ import com.milaboratory.core.sequence.NucleotideSequence
...
-    @Option(
-        description = ["Read pre-processing: trimming quality threshold"],
-        names = ["--trimming-quality-threshold"]
-    )
-    var trimmingQualityThreshold: Byte = 0 // 17
+            @Option(
+                description = ["Read tag pattern from a file."],
+                names = ["--tag-pattern-file"]
+            )
+            var tagPatternFile: String? = null
```
Comment (gnefedev) [inline src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt:161]: you can use Path type

File: src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt [hunk 3]
```diff
@@ -37,758 +35,743 @@ import com.milaboratory.core.sequence.NucleotideSequence
...
-            "file_R1.(fastq[.gz]|fasta|bam|sam) [file_R2.(fastq[.gz]|bam|sam)] [file_RN.(bam|sam)] alignments.vdjca",
-            "Use {{n}} if you want to concatenate files from multiple lanes, like:",
-            "my_file_L{{n}}_R1.fastq.gz my_file_L{{n}}_R2.fastq.gz"
-        ]
+object CommandAlign {
+    const val COMMAND_NAME = "align"
+
+    data class Params(
+        @JsonProperty("species") val species: String = "",
+        @JsonProperty("libraryName") val libraryName: String = "default",
```
Comment (gnefedev) [inline src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt:77]: why do you set default value? this constructor never executed

File: src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt [hunk 4]
```diff
@@ -37,758 +35,743 @@ import com.milaboratory.core.sequence.NucleotideSequence
...
+        description = ["Builds alignments with V,D,J and C genes for input sequencing reads."]
     )
-    var writeAllResults = false
+    class Cmd : CmdBase() {
+        @Option(
+            description = ["Analysis preset. Sets all significant parameters of this and all downstream analysis steps. " +
+                    "This is a required parameter. It is very important to carefully select the most appropriate preset " +
+                    "for the data you analyse."], names = ["-p", "--preset"]
+        )
+        lateinit var presetName: String
```
Comment (gnefedev) [inline src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt:240]: required?

File: src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt (hunk 3 above)
Comment (dbolotin) [reply src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt:77]: Yap. Will remove this.

File: src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt (hunk 4 above)
Comment (dbolotin) [reply src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt:240]: Already did, no committed yet.

**Pull request closed without merging**